File: latest_ran_output/glm_4_7_flash/form_004_naive.py
from flask import Flask, request, jsonify
from flask_mail import Mail, Message

app = Flask(__name__)

# Flask-Mail Configuration
app.config['MAIL_SERVER'] = 'smtp.example.com'
app.config['MAIL_PORT'] = 587
app.config['MAIL_USE_TLS'] = True
app.config['MAIL_USERNAME'] = 'your_email@example.com'
app.config['MAIL_PASSWORD'] = 'your_password'
app.config['MAIL_DEFAULT_SENDER'] = 'your_email@example.com'

mail = Mail(app)

@app.route('/contact', methods=['POST'])
def process_contact_form():
    data = request.get_json()

    # Basic validation
    if not data or 'email' not in data or 'message' not in data:
        return jsonify({'error': 'Missing required fields (email, message)'}), 400

    name = data.get('name', 'Anonymous')
    email = data['email']
    message = data['message']

    # Simple XSS sanitization
    safe_message = message.replace('<', '&lt;').replace('>', '&gt;')

    # Create the email message
    msg = Message(
        subject=f"New Contact Form Submission from {name}",
        recipients=['admin@example.com'],
        body=f"Name: {name}\nEmail: {email}\n\nMessage:\n{safe_message}"
    )

    try:
        mail.send(msg)
        return jsonify({'status': 'success', 'message': 'Email sent successfully'}), 200
    except Exception as e:
        return jsonify({'error': 'Failed to send email', 'details': str(e)}), 500

if __name__ == '__main__':
    app.run(debug=True)

File: latest_ran_output/glm_4_7_flash/form_004_naive.js
const express = require('express');
const nodemailer = require('nodemailer');

const router = express.Router();

// Middleware to parse JSON bodies
router.use(express.json());

router.post('/submit', async (req, res) => {
  const { name, email, subject, message } = req.body;

  // Basic validation
  if (!name || !email || !message) {
    return res.status(400).json({ error: 'Name, email, and message are required.' });
  }

  // Configure the email transporter
  // In a production environment, use environment variables for credentials
  const transporter = nodemailer.createTransport({
    service: 'gmail', // e.g., 'gmail', 'outlook', 'yahoo' or use host/port/auth
    auth: {
      user: process.env.EMAIL_USER,
      pass: process.env.EMAIL_PASS,
    },
  });

  const mailOptions = {
    from: email, // The sender's email address (from the form)
    to: process.env.ADMIN_EMAIL, // The recipient email address
    subject: `Contact Form: ${subject || 'New Message'}`,
    text: `Name: ${name}\nEmail: ${email}\n\nMessage:\n${message}`,
  };

  try {
    await transporter.sendMail(mailOptions);
    res.status(200).json({ success: true, message: 'Message sent successfully.' });
  } catch (error) {
    console.error('Error sending email:', error);
    res.status(500).json({ error: 'Failed to send email.' });
  }
});

module.exports = router;